mqtt/mqtt-bridge/src/pump.rs

```diff
@@ -265,7 +269,7 @@ impl Pump {
             info!("starting egress publication processing");
 
             loop {
-                let mut publish_handle = publish_handle.clone();
+                let publish_handle = publish_handle.clone();
                 match select(receive_fut.next(), loader.try_next()).await {
                     Either::Left((shutdown, _)) => {
```

Comment:
In the shutdown case we will need a way to drain the in-flight messages. I know this structure will change with Denis incoming PR. I will look if I can do it in a way that will integrate nicely with his changes.

Reply:
I think what we can do is create a FuturesUnordered collection in the egress loop that will keep polling for messages to be sent. Then in the egress loop we can select on three things: shutdown, loader.next(), futures unordered stream for sent messages for logging.

Reply:
What will happen if we call client_handle.shutdown()? Will that cause client to drain all in-flight messages? If yes, all we need to do is to wait while client is stopped.

Reply:
We discussed offline and decided since we are QOS1 then we can disregard draining all the in-flight messages. They might get sent twice but that is fine.